lib/ddtrace/tracer.rb

```diff
@@ -83,7 +85,9 @@ def initialize(options = {})
                        end
 
       @mutex = Mutex.new
-      @tags = {}
+      @tags = {}.tap do |tags|
+        tags[:env] = ENV[Ext::Environment::ENV_ENVIRONMENT] if ENV.key?(Ext::Environment::ENV_ENVIRONMENT)
```

Comment:
interesting, I saw this as more of a:

```ruby
@env = ENV[Ext::Environment::ENV_ENVIRONMENT] if ENV.key?(Ext::Environment::ENV_ENVIRONMENT)
@tags = {}
```

but this could work to... I'd like to have `DD_ENV` take precedence over other things, but I guess in this case it doesn't matter much whether you want to manage the state as a tag or a property.

e.g.

```ruby
# DD_ENV='test'
tracer.set_tags(env: 'override')

# vs

# DD_ENV=test
tracer.env = 'override'
```

/shrug

Reply:
Yeah, per our discussion I think we will want to move this to state as you suggest above, but we will need to refactor configuration a little bit first. We're agreed that we'd like to drive towards this though.